pkg/manifests/manifests.go

```diff
@@ -142,9 +169,52 @@ func (f *Factory) RouterDeployment(cr *ingressv1alpha1.ClusterIngress) (*appsv1.
 	}
 	deployment.Spec.Selector.MatchLabels["router"] = name
 
+	statsSecretName := fmt.Sprintf("router-stats-%s", cr.Name)
 	env := []corev1.EnvVar{
 		{Name: "ROUTER_SERVICE_NAME", Value: cr.Name},
-	}
+		{Name: "STATS_USERNAME", ValueFrom: &corev1.EnvVarSource{
+			SecretKeyRef: &corev1.SecretKeySelector{
+				LocalObjectReference: corev1.LocalObjectReference{
+					Name: statsSecretName,
+				},
+				Key: "statsUsername",
+			},
+		}},
+		{Name: "STATS_PASSWORD", ValueFrom: &corev1.EnvVarSource{
+			SecretKeyRef: &corev1.SecretKeySelector{
+				LocalObjectReference: corev1.LocalObjectReference{
+					Name: statsSecretName,
+				},
+				Key: "statsPassword",
+			},
+		}},
+	}
+
+	// Enable prometheus metrics
+	certsSecretName := fmt.Sprintf("router-metrics-certs-%s", cr.Name)
+	certsVolumeName := "metrics-certs"
+	certsVolumeMountPath := fmt.Sprintf("/etc/pki/tls/%s", certsSecretName)
```

Comment:
Do we really want the mount path to have the router's name? If we do, we should use `filepath.Join` to compose the path.  However, it would be simpler to use a constant path name, and I don't see a reason not to.